### PR TITLE
Fixes Escape Pods

### PR DESCRIPTION
**Pull request by The-Moon-Itself**
## About The Pull Request
Escape pods now have floors again!

All maps from the same map file sharing a single map template datum caused some issues when trying to load multiple of the same map at a same time, specifically the cached_map var only stores the most recent and therefore most likely to not be finished yet map parser. This caused issues when placing skipover turfs on shuttles that expected that cache to not be still generating. I changed it so that each map now uses its assigned map parser instead of the most recent one of its type to use for shuttle skipover placement.

## Why It's Good For The Game
Escape pods should help you escape alive, not cause you to fall into space 

## Testing Photographs and Procedure

<details>
<summary>Pods working on MetaStation</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/51838176/e68834a8-8c47-4360-a656-8be1cb4a660a

</details>

## Changelog
:cl:
fix: Escape pods are fixed
/:cl:

### CODE REVIEW

**Comment by The-Moon-Itself** [comment]
Okay, turns out this has been identified before in #9259, which explicitly states that the solution I did may not be the best fix, but I don't really see why. I guess in that it's also referring to ruin generation which I don't know the first thing about. Maybe I should take a look at this again.

**Comment by The-Moon-Itself** [comment]
I think I'm just going to leave this as it is. I don't want to jump down another rabbit hole and try to tackle code written by multiple people who know more of what they're doing than I do, leading into another burnout. At the end of the day I'm just some guy who got all his programming knowledge from tinkering in roblox, minecraft computercraft, and empty unity projects with no academic CS education at all, and while I can still do some cool things, I need to accept my limits. I just need to accept that these are pull *requests*, not obligations. I'm just offering what I've made with the skills I have, and I'm not letting anyone down if the maintainers choose if anything I put up here isn't worth adding along with any technical debt that I might unwittingly include in the code.

**Review by PowerfulBacon — APPROVED** [review]
Seems okay, I thought the cache was set after it was fully generated but maybe not

**Pull request merged**
